docs/tutorial/Tutorial.md

```diff
@@ -0,0 +1,501 @@
+# MMTk Tutorial
+
+In this tutorial, you will build multiple garbage collectors using MMTK from scratch. 
+
+This tutorial is a work in progress. Some sections may be rough, and others may be missing information (especially about import statements). If something seems to be missing, it may be worth looking at the relevant section the finished SemiSpace garbage collector.
+
+
+## Contents
+* [Introduction](#introduction)
+* [Preliminaries](#preliminaries)
+  * [Set up MMTK and OpenJDK](#set-up-mmtk-and-openjdk)
+    * [Basic set up](#basic-set-up)
+    * [Set up benchmarks](#set-up-benchmarks)
+    * [Working with multiple VM builds](#working-with-multiple-vm-builds)
+  * [Create MyGC](#create-mygc)
+* [Building a Semispace Collector](#building-a-semispace-collector)
+* [Further Reading](#further-reading)
+
+
+## Introduction
+### What *is* the MMTk?
+The Memory Management Toolkit (MMTk) is a framework to design and implement memory managers. It has a core (this repository) written in Rust, and bindings that allow it to work with OpenJDK, V8, and JikesRVM, with more bindings currently in development. The toolkit has a number of pre-built collectors, and is intended to make it relatively simple to expand upon or build new collectors. Many elements common between collectors can be easily implemented.
+
+### What will this tutorial be covering?
+This tutorial is intended to get you comfortable with building garbage collectors in the MMTk.
+
+You will first be guided through building a Semispace collector. After that, you will extend this collector in various ways that are not particularly practical, but introduce different concepts implemented in the MMTk. These exersizes will be less guided, but hints and functional solutions will be available in case you get stuck. There will also be questions at various points in the tutorial, intended to encourage you to think about what the code is *doing* and potentially motivate further research.
```

Comment:
Ah, I wrote that when I was thinking of putting more exercises (in the style of the "triplespace" exercise) after the Semispace section rather than going through the GenCopy collector (I don't remember exactly why). I'll fix the wording in the next commit.